### ISSUE TITLE
IfcPolygonalFaceSet is not supported yet?

### ISSUE THREAD

**Issue by I-Sokolov** [xBimTeam/XbimEssentials#116]
ExpressType(IFCINDEXEDPOLYGONALFACE) returns NULL

Xbim.Common.Exceptions.XbimParserException: Parser failed on line 0, column 0 ---> System.NullReferenceException:
   в Xbim.Common.Metadata.ExpressMetaData.ExpressTypeId(String typeName)  C:\BuildAgent\work\77d7b8e93613846d\Xbim.Common\Metadata\ExpressMetaData.cs:  ln270

**Comment by I-Sokolov** [comment]
[Beam_01_ABD.ifc.txt](https://github.com/xBimTeam/XbimEssentials/files/1430185/Beam_01_ABD.ifc.txt)

**Comment by CBenghi** [comment]
Hello @I-Sokolov,

IfcPolygonalFaceSet is a new entity in IFC4 Addendum 2.
Our scheme is still using addendum 1 at the moment.
We could start a conversation about ways to live with more than one... it's more complicated than it seems.

Best,
Claudio

**Comment by I-Sokolov** [comment]
Correct, it is IFC 4 Add 2.
IfcPolygonalFace is a part of IFC 4 Reference View while IfcFacetedBrep and Boolean (even clipped) solids are not allowed. 
Thus I expect nearly all IFC files from certified vendors will use IfcPolygonalFace. 
Also note IfcIdexedPolyline is only way for arcs in IFC4 RV.

**Comment by CBenghi** [comment]
You have a valid point...
[Brain cells start crunching noise]

**Comment by I-Sokolov** [comment]
Have you any update on the issue?
Is this in plans and time expectation?
Thanks and Happy New Year

**Comment by CBenghi** [comment]
We have started work on this.
The first step is completed as we have progressed develop branch to Add-2 schema, but I still have to engage in the geometry engine updates. 
I don't see time to engage further until the end of the month as I have a few deadlines by then.
PRs are welcome (I realise that dealing with xbim.geometry might be a little intimidating, but I'd be happy to help if you do).
Claudio
